### DISCUSSION TITLE
simple definitions

### DISCUSSION THREAD

**Discussion by MrAmbiG** [mrdbourke/pytorch-deep-learning#1010] [Show and tell]
- scalar: (0 dimentional) representation of magnitude
- vector: 1 dimentional representation of magnitude & direction
- matrix: 2 dimentional representation of magnitude & direction
- tensor: n dimentional representation of magnitude & direction

**Comment by ghost** [comment]
1. **Scalar**: Represents a single numerical value, typically associated with magnitude (e.g., temperature, mass, speed) but without any direction. It can be considered as a 0-dimensional object because it doesn't have direction.

2. **Vector**: Represents both magnitude and direction in space. It is a 1-dimensional object because it has a direction associated with it. Examples include velocity, force, and displacement.

3. **Matrix**: Represents a rectangular array of numbers arranged in rows and columns. It is a 2-dimensional object because it has both rows and columns, and each element can represent some quantity related to direction and magnitude, depending on the context.

4. **Tensor**: Extends the concept of vectors and matrices to higher dimensions. It is an n-dimensional array of numbers, where n can be any positive integer. Tensors are used to represent data in many dimensions and are common in fields such as physics, engineering, and machine learning. They generalize the concepts of scalars, vectors, and matrices.
